internal/manager/application/application.go

```diff
@@ -590,6 +592,38 @@ func (m *ApplicationManager) RemoveFinalizers(ctx context.Context, incoming *v1a
 	return updated, err
 }
 
+func (m *ApplicationManager) List(ctx context.Context, selector backend.ApplicationSelector) ([]v1alpha1.Application, error) {
+	return m.applicationBackend.List(ctx, selector)
+}
+
+// RevertManagedAppChanges compares the actual spec with expected spec stored in cache,
+// if actual spec doesn't match with cache, then it is reverted to be in sync with cache, which is same as principal.
+func (m *ApplicationManager) RevertManagedAppChanges(ctx context.Context, app *v1alpha1.Application) bool {
+	logCtx := log().WithFields(logrus.Fields{
+		"component":       "RevertManagedAppChanges",
+		"application":     app.QualifiedName(),
+		"resourceVersion": app.ResourceVersion,
+	})
+
+	sourceUID, exists := app.Annotations[manager.SourceUIDAnnotation]
+	if exists && m.mode == manager.ManagerModeManaged {
+		if cachedAppSpec, ok := appCache.GetApplicationSpec(ty.UID(sourceUID), logCtx); ok {
+			logCtx.Debugf("Application: %s is available in agent cache", app.Name)
+
+			if diff := reflect.DeepEqual(cachedAppSpec, app.Spec); !diff {
+				app.Spec = cachedAppSpec
+				if _, err := m.UpdateManagedApp(ctx, app); err != nil {
```

Comment:
Can you add a log statement (at Info level) just before the call to `m.UpdateManagedApp`, indicating that we are about to revert the Application?

Reply:
it is done, PTAL